internal/configs/hcl2shim/mock_value_composer.go

```diff
@@ -0,0 +1,353 @@
+package hcl2shim
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+
+	"github.com/opentofu/opentofu/internal/configs/configschema"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// ComposeMockValueBySchema composes mock value based on schema configuration. It uses
+// configuration value as a baseline and populates null values with provided defaults.
+// If the provided defaults doesn't contain needed fields, ComposeMockValueBySchema uses
+// its own defaults. ComposeMockValueBySchema fails if schema contains dynamic types.
+func ComposeMockValueBySchema(schema *configschema.Block, config cty.Value, defaults map[string]cty.Value) (
+	cty.Value, tfdiags.Diagnostics) {
+	return mockValueComposer{}.composeMockValueBySchema(schema, config, defaults)
+}
+
+type mockValueComposer struct {
+	getMockStringOverride func() string
+}
+
+func (mvc mockValueComposer) getMockString() string {
+	f := getRandomAlphaNumString
+
+	if mvc.getMockStringOverride != nil {
+		f = mvc.getMockStringOverride
+	}
+
+	return f()
+}
+
+func (mvc mockValueComposer) composeMockValueBySchema(schema *configschema.Block, config cty.Value, defaults map[string]cty.Value) (cty.Value, tfdiags.Diagnostics) {
+	var configMap map[string]cty.Value
+	var diags tfdiags.Diagnostics
+
+	if !config.IsNull() {
+		configMap = config.AsValueMap()
+	}
+
+	impliedTypes := schema.ImpliedType().AttributeTypes()
+
+	mockAttrs, moreDiags := mvc.composeMockValueForAttributes(schema, configMap, defaults)
+	diags = diags.Append(moreDiags)
+	if moreDiags.HasErrors() {
+		return cty.NilVal, diags
+	}
+
+	mockBlocks, moreDiags := mvc.composeMockValueForBlocks(schema, configMap, defaults)
+	diags = diags.Append(moreDiags)
+	if moreDiags.HasErrors() {
+		return cty.NilVal, diags
+	}
+
+	mockValues := mockAttrs
+	for k, v := range mockBlocks {
+		mockValues[k] = v
+	}
+
+	for k := range defaults {
+		if _, ok := impliedTypes[k]; !ok {
+			diags = diags.Append(tfdiags.WholeContainingBody(
+				tfdiags.Warning,
+				fmt.Sprintf("Ignored mock/override field `%v`", k),
+				"The field is unknown. Please, ensure it is a part of resource definition.",
+			))
+		}
+	}
+
+	return cty.ObjectVal(mockValues), diags
+}
+
+func (mvc mockValueComposer) composeMockValueForAttributes(schema *configschema.Block, configMap map[string]cty.Value, defaults map[string]cty.Value) (map[string]cty.Value, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
+
+	addPotentialDefaultsWarning := func(key, description string) {
+		if _, ok := defaults[key]; ok {
+			diags = diags.Append(tfdiags.WholeContainingBody(
+				tfdiags.Warning,
+				fmt.Sprintf("Ignored mock/override field `%v`", key),
+				description,
+			))
+		}
+	}
+
+	mockAttrs := make(map[string]cty.Value)
+
+	impliedTypes := schema.ImpliedType().AttributeTypes()
+
+	for k, attr := range schema.Attributes {
+		// If the value present in configuration - just use it.
```

Comment:
I want to validate I understand this scenario correctly -
From what I understand if I have the resource:
```
resource "something" "hi" {
    id = "2342"
}
```
If I have an "override_resource" block for this resource, I won't be able to override the `id` attribute?